src/main/java/org/wso2/carbon/extension/analytics/publisher/file/FileEventAdapter.java

```diff
@@ -0,0 +1,142 @@
+import java.nio.channels.FileChannel;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FileEventAdapter implements OutputEventAdapter {
+    private static final Log log = LogFactory.getLog(FileEventAdapter.class);
+    private OutputEventAdapterConfiguration eventAdapterConfiguration;
+    private Map<String, FileChannel> fileChannelMap = new HashMap<>();
+
+    public FileEventAdapter(OutputEventAdapterConfiguration eventAdapterConfiguration) {
+        this.eventAdapterConfiguration = eventAdapterConfiguration;
+    }
+
+    /**
+     * This method is called when initiating event publisher bundle.
+     * Relevant code segments which are needed when loading OSGI bundle can be included in this method.
+     *
+     * @throws OutputEventAdapterException
+     */
+    public void init() throws OutputEventAdapterException {
+
+    }
+
+    /**
+     * This method is used to test the connection of the publishing server.
+     *
+     * @throws TestConnectionNotSupportedException
+     * @throws ConnectionUnavailableException
+     */
+    public void testConnect() throws TestConnectionNotSupportedException, ConnectionUnavailableException {
+
+    }
+
+    /**
+     * Can be called to connect to back end before events are published.
+     *
+     * @throws ConnectionUnavailableException
+     */
+    public void connect() throws ConnectionUnavailableException {
+
+    }
+
+    /**
+     * Publish events. Throws ConnectionUnavailableException if it cannot connect to the back end.
+     *
+     * @param message           Message object.
+     * @param dynamicProperties Dynamic properties list.
+     * @throws ConnectionUnavailableException
+     */
+    public void publish(Object message, Map<String, String> dynamicProperties) throws ConnectionUnavailableException {
+        ByteBuffer buffer = null;
+        String file = dynamicProperties.get(FileEventAdapterConstants.ADAPTER_FILE_NAME);
+        try {
+            buffer = ByteBuffer.allocate(Integer.parseInt(dynamicProperties
+                    .get(FileEventAdapterConstants.ADAPTER_BUFFER_SIZE)));
+            buffer.put(message.toString().getBytes());
+            buffer.flip();
+            if (fileChannelMap.get(file) == null) {
+                createNewChannel(file);
+            }
+            fileChannelMap.get(file).write(buffer);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        } finally {
+            if (buffer != null) {
+                buffer.clear();
+            }
+        }
+    }
+
+    /**
+     * Will be called after publishing is done, or when ConnectionUnavailableException is thrown.
+     */
+    public void disconnect() {
+        try {
+            for (String keyList : fileChannelMap.keySet()) {
+                fileChannelMap.get(keyList).close();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error occurred while close the FileChannel", e);
```

Comment:
Actually, above code needs to be corrected as below..

for (String keyList : fileChannelMap.keySet()) {
try{
            fileChannelMap.get(keyList).close();
} catch (IOException e){
log.error( ERROR MESSAGE, e);
}
           }